https://issues.redhat.com/browse/ACM-5962 Known issue pruneObjectBeha… (#5204)
open-cluster-management/rhacm-docs: release_notes/known_issues_governance.adoc

The configuration policy resources are automatically removed from the cluster and the CRD exits its terminating state. If the add-on has already been reinstalled, the CRD is recreated automatically without a deletion timestamp.

[#pruneobjbeh-not-working-existing-config-policy]
== _pruneObjectBehavior_ does not work when modifying existing configuration policy
//2.6:25261
//2.7+2.8:5939

When you modify an existing configuration policy, `pruneObjectBehavior` does not work. View the following reasons why `pruneObjectBehavior` might not work:

> - If you set `pruneObjectBehvior` to `DeleteAll` or `DeleteIfCreated`, old resources that were created before modifying are not cleaned correctly. Only new resources from policy creations and policy updates are tracked and deleted when you delete the configuration policy.

Reviewer: ```suggestion
- If you set `pruneObjectBehavior` to `DeleteAll` or `DeleteIfCreated` in a configuration policy, old resources that were created before modifying are not cleaned correctly. Only new resources from policy creations and policy updates are tracked and deleted when you delete the configuration policy.
```

Reviewer: What are we modifying here? "...resources that you created before modifying the `pruneObjectBehavior` setting are not cleaned correctly"?

Author: Modifying the configuration policy

Author: @cadawson I edited your suggestion, did that clarify things?

Reviewer: Yep. That works for me. Thanks!

> - If you set `pruneObjectBehvior` to `None` or fail to set the parameter value, old objects might be unintentionally deleted on the managed cluster. Specifically, this occurs when a user changes the `name`, `namespace`, `kind`, or `apiversion` in the template. The parameter fields can dynamically change when the `object-templates-raw` or `nameSelector` parameters change. 

Reviewer 2: should be `namespaceSelector`?!!

Author: Great catch @yiraeChristineKim changing that now

Reviewer: @dockerymick: Please make sure you add the missing "a" to "Behavior".

Author: Updated

> - If you set `pruneObjectBehvior` to `None` or fail to set the parameter value, old objects might be unintentionally deleted on the managed cluster. Specifically, this occurs when a user changes the `name`, `namespace`, `kind`, or `apiversion` in the template. The parameter fields can dynamically change when the `object-templates-raw` or `nameSelector` parameters change. 

Reviewer 2: To me, **fail to set the parameter value** unclear, what does the **parameter value** indicate?

Reviewer 2: is that `name`, `namespace`, `kind`, or `apiversion`?

Author: From my perspective, the parameter values for `pruneObjectBehavior` are `None`, `DeleteAll`, and `DeleteIfCreated`

> - If you set `pruneObjectBehvior` to `None` or fail to set the parameter value, old objects might be unintentionally deleted on the managed cluster. Specifically, this occurs when a user changes the `name`, `namespace`, `kind`, or `apiversion` in the template. The parameter fields can dynamically change when the `object-templates-raw` or `nameSelector` parameters change. 

Reviewer 3: ```suggestion
- If you set `pruneObjectBehvior` to `None` or fail to set the parameter value, old objects might be unintentionally deleted on the managed cluster. Specifically, this occurs when a user changes the `name`, `namespace`, `kind`, or `apiversion` in the template. The parameter fields can dynamically change when the `object-templates-raw` or `namespaceSelector` parameters change. 
```

@yiraeChristineKim is keeping me honest and noticed this typo I missed

> - If you set `pruneObjectBehvior` to `None` or fail to set the parameter value, old objects might be unintentionally deleted on the managed cluster. Specifically, this occurs when a user changes the `name`, `namespace`, `kind`, or `apiversion` in the template. The parameter fields can dynamically change when the `object-templates-raw` or `nameSelector` parameters change. 

Reviewer: ```suggestion
- If you set `pruneObjectBehavior` to `None` or do not set the parameter value, old objects might be unintentionally deleted on the managed cluster. Specifically, this occurs when you change the `name`, `namespace`, `kind`, or `apiversion` in the template. The parameter fields can dynamically change when the `object-templates-raw` or `nameSelector` parameters change. 
```

Author: Thanks! I updated this
